sources/packages/backend/apps/api/src/services/disbursement-schedule/_tests_/unit/disbursement-schedule.getCOEApprovalPeriodStatus.spec.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  NotificationActionsService,
+  SequenceControlService,
+} from "@sims/services";
+import { DisbursementScheduleService } from "../../disbursement-schedule-service";
+import { DataSource } from "typeorm";
+import { addDays } from "@sims/utilities";
+import { COE_WINDOW } from "../../../../utilities";
+import { COEApprovalPeriodStatus } from "../../disbursement-schedule.models";
+
+describe("DisbursementScheduleService-getCOEApprovalPeriodStatus", () => {
+  let service: DisbursementScheduleService;
+
+  beforeAll(async () => {
+    const dataSource = {} as DataSource;
+    dataSource.getRepository = jest.fn();
+    const sequenceService = {} as SequenceControlService;
+    const notificationActionsService = {} as NotificationActionsService;
+    service = new DisbursementScheduleService(
+      dataSource,
+      sequenceService,
+      notificationActionsService,
+    );
+  });
+
+  it("Should throw an error when the disbursement date is undefined.", () => {
+    // Arrange
+    const disbursementDate = undefined;
+    const studyEndDate = new Date();
+
+    // Act and Assert.
+    expect(() => {
+      service.getCOEApprovalPeriodStatus(disbursementDate, studyEndDate);
+    }).toThrow(
+      "disbursementDate and studyEndDate are required for COE window verification.",
+    );
+  });
+
+  it("Should throw an error when both the disbursement date and study end date are undefined.", () => {
+    // Arrange
+    const disbursementDate = undefined;
+    const studyEndDate = undefined;
+
+    // Act and Assert.
+    expect(() => {
+      service.getCOEApprovalPeriodStatus(disbursementDate, studyEndDate);
```

Comment:
Suggestion. we can avoid creating constant to pass undefined.

Reply:
mm, I thought this is more clear to the person looking to test. like now will understand the first parameter is `disbursementDate `and its null